pkg/ebpf/c/tracer-ebpf.h

```diff
@@ -56,6 +56,72 @@ typedef struct {
     tcp_stats_t tcp_stats;
 } tcp_conn_t;
 
+#define HTTP_BUFFER_SIZE 15
+#define HTTP_STATUS_CODE_SIZE 3
+#define HTTP_BATCH_SIZE 5
+// The greater this number is the less likely are colisions/data-races between the flushes
+#define HTTP_BATCH_PAGES 5
+
+// From include/net/tcp.h
+// tcp_flag_byte(th) (((u_int8_t *)th)[13])
+#define TCP_FLAGS_OFFSET 13
+#define TCPHDR_FIN 0x01
+
+// Data generated from parsing a skb struct
+typedef struct {
+    conn_tuple_t tup;
+    __u32 data_off;
+    __u32 data_end;
+    __u8 tcp_flags;
+} skb_info_t;
+
+typedef enum {
+    HTTP_UNKNOWN           = 0,
+    HTTP_RESPONDING        = 1,
+    HTTP_REQUESTING_GET    = 1 << 1,
+    HTTP_REQUESTING_POST   = 1 << 2,
+    HTTP_REQUESTING_PUT    = 1 << 3,
+    HTTP_REQUESTING_DELETE = 1 << 4,
+    HTTP_REQUESTING_HEAD   = 1 << 5,
+} http_state_t;
```

Comment:
This feels confusing, as it is mixing two concerns:
1. the state of the transaction: unknown->requesting->responding
2. The request method

It is also referenced in code as both `state` and `packet_type`. Perhaps they should be separate things, or at least different masked parts of the byte.

Method also doesn't need to be a bitmask, since it cannot be multiple methods at the same time.